### PR TITLE
Update app.exe matching to work with both public and beta Talon

### PR DESCRIPTION
**Pull request by nriley**
- Update mixed-case app.exe: matchers to work with public and beta Talon.
- Update app switcher to work with public and beta Talon.

### CODE REVIEW

**Comment by nriley** [comment]
I've not got Windows at home so this specific set of changes aren't well tested, but wanted to get it out there since these changes will affect Talon beta users. Will test more on Monday if nobody else does over the weekend!

File: apps/adobe/adobe_acrobat_reader_dc.py
Comment (fidgetingbits) [inline apps/adobe/adobe_acrobat_reader_dc.py:1]: These two are missing the escaped dot

Comment (nriley) [reply apps/adobe/adobe_acrobat_reader_dc.py:1]: Thanks. I've got more changes not yet moved over to this branch but will be the weekend until I get back to it.

**Comment by knausj85** [comment]
Should we update the copy context commands to do this regex magic?

**Comment by nriley** [comment]
> Should we update the copy context commands to do this regex magic?

Good call, yes! Will do.

**Comment by AndreasArvidsson** [comment]
Definitely. We should also update create app context:
https://github.com/talonhub/community/blob/c6e1969ab516d509f51b9c8acd01bd35d2262d28/plugin/talon_helpers/create_app_context.py#L79

We should probably even unify the implementation of these two

**Comment by nriley** [comment]
> Definitely. We should also update create app context:
> 
> https://github.com/talonhub/community/blob/c6e1969ab516d509f51b9c8acd01bd35d2262d28/plugin/talon_helpers/create_app_context.py#L79

Done. 

> We should probably even unify the implementation of these two

Agreed, although would prefer to do another time. They generate Windows app matching differently (create_app_context just matches app.exe and talon copy context also matches app.name).

File: core/app_switcher/app_switcher.py [hunk 1]
```diff
@@ -23,6 +25,9 @@
 # a list of the current overrides
 overrides = {}
 
+# apps to exclude from running list
+excludes = set()
```
Comment (knausj85) [inline core/app_switcher/app_switcher.py:29]: Probably worth adding a comment that any line without a comma is excluded? This is a bit different from several of the other CSVs, where it's assumed the spoken form and result are the same in this case.

**Review by knausj85 — APPROVED** [review]
This look solid to me with one minor comment

File: core/app_switcher/app_switcher.py (hunk 1 above)
Comment (nriley) [reply core/app_switcher/app_switcher.py:32]: Agreed, that was a bit too mysterious (and comments in this file were out of date). Updated.

**Pull request merged**